cmd/zoekt-sourcegraph-indexserver/index_test.go

```diff
@@ -211,7 +262,12 @@ func TestIndex(t *testing.T) {
 				return nil
 			}
 
-			if err := gitIndex(&tc.args, runCmd); err != nil {
+			c := gitIndexConfig{
+				runCmd:                runCmd,
+				getRepositoryMetadata: tc.mockGetRepositoryMetadata,
+			}
```

Comment:
style: I like the "static" arguments to come before the variable arguments. This style is influenced by haskell/etc where currying is popular. In fact I would recommend moving to:

````
type gitIndexer struct {
   // mockable functions
}

func (*gitIndexer) Index(args) ...
```

Edit: reading the existing call site it is probably better kept as the first argument.